### ISSUE TITLE
Class incorrectly implements interface containing methods with optional parameters

### ISSUE THREAD

**Issue by ThomasKiesel** [microsoft/TypeScript#61380]
### 🔎 Search Terms

interface implementation optional parameter

### 🕗 Version & Regression Information

-We detected the problem in Typescript 5.7.2. 
- According to the playground, it already exists since Typescript 3.3.3

### ⏯ Playground Link

https://www.typescriptlang.org/play/?ts=5.7.3#code/JYOwLgpgTgZghgYwgAgJLIN4FgBQBIAWwjAAsB7AEwAo4AuZAZzClAHMAaZAIwH56mWIVgEpaANzLAKAblwBfXLgQAbOAwbIAwsmAEADsohFwG9NnxFSlGv2ZtOXW4JH0JUzLjx4EZEAzKGAHTKZKxUXIEMAK5cAmxUAIzCwp4KOGm4YnBQyAj02gC8yCAQAO5aVMKyOAiBluTUAEQJjVXIeAD0HVpk+sCGOVBwwAwQGtBQZFD0AKIAHnoQCJAUyABMyNmsUcZgGlxRYMisZEcJgYo4WTnA9OhFJeWaldXAdcQNVM2t0p3dmr09P1oMgKGQxsVTsghiMUHAQMgJlNAsh5ktDsBfMgyDBkKQUD4KChDHAKBowGRNgiIHMkHowJiQLggA

### 💻 Code

interface I {
	method(a: string, b?: string):void;
}

class C implements I {
	method(a: string, b: string): void {
		console.log(b.substring(1))
	}
}

var c: C = new C();
c.method("1"); 	// Compiler raises error: Expected 2 arguments but got 1.

var i: I = new C();
i.method("1");	// Compiler does not raise an error. Execution of the code leads to an exception

### 🙁 Actual behavior

The Typescript compiler seems to conclude that class C implements interface I correctly, although the signature of the "method" in interface I and class C are not equivalent.

Execution of the last line of code leads to an exception, since the parameter "b" in "method" is undefined. The implementation of "method" does not expect the parameter "b" to be undefined.

### 🙂 Expected behavior

Typescript shall report an error that class C does not implement interface I correctly, since the method signatures in class C and interface I are not compatible. This would prevent the exception during runtime.

**Comment by jcalz** [comment]
This is working as intended, [method parameters are bivariant](https://www.typescriptlang.org/docs/handbook/type-compatibility.html#function-parameter-bivariance), for better or worse. You can change your method to function-property syntax and get the error you expect (assuming you use `--strictFunctionTypes`):

```ts
interface I {
	method: (a: string, b?: string) => void;
}
class C implements I {
	method(a: string, b: string): void {
		console.log(b.substring(1))
	}
}
var c: C = new C();
c.method("1");
var i: I = new C();
i.method("1");
```

See #46591, #60512, #18654 and

**Comment by ThomasKiesel** [comment]
Thank you very much for your fast reaction and your solution to handle this issue.
